web-console/src/lib/components/home/Pipelines.tsx

```diff
@@ -2,159 +2,97 @@
 
-import Box from '@mui/material/Box'
 import Card from '@mui/material/Card'
 import CardContent from '@mui/material/CardContent'
 import CardHeader from '@mui/material/CardHeader'
-import { useTheme } from '@mui/material/styles'
+import { Theme, useTheme } from '@mui/material/styles'
 import Typography from '@mui/material/Typography'
-
-const series = [
-  [{ data: [19, 58, 45, 11] }],
-  [{ data: [12, 58, 45, 13] }],
-  [{ data: [30, 40, 50, 55] }],
-  [{ data: [1, 2, 2, 2] }]
-]
-
-interface DataType {
-  name: string
-  totalTput: number
-  chartValue: number
-  designation: string
-  chartColor?: ThemeColor
-}
-
-const data: DataType[] = [
-  {
-    chartValue: 85,
-    name: 'Tech Averages',
-    totalTput: 138,
-    chartColor: 'secondary',
-    designation: 'Calculate 30 day rolling...'
-  },
-  {
-    name: 'S&P 500 Pairs',
-    chartValue: 70,
-    totalTput: 149,
-    chartColor: 'warning',
-    designation: 'Finds pairs in S&P500 stocks.'
-  },
-  {
-    name: 'NYSE <> CME Arbitrage',
-    chartValue: 25,
-    totalTput: 160,
-    designation: 'Exploit price differences...'
-  },
-  {
-    name: 'Mean Reversion',
-    chartValue: 75,
-    totalTput: 166,
-    chartColor: 'error',
-    designation: 'Find tickers outside...'
-  }
-]
+import { useQuery } from '@tanstack/react-query'
+import { PipelineManagerQuery } from '$lib/services/defaultQueryFn'
+import { PipelineThumb } from '$lib/components/home/pipelines/PipelineThumb'
+import { PipelineStatus } from 'src/lib/services/manager'
+import { partition } from 'ts-practical-fp'
+import { ListItem } from '@mui/material'
 
 const Pipelines = () => {
   const theme = useTheme()
+  const apexOptions = apexChartsOptions(theme)
 
-  const options: ApexOptions = {
-    chart: {
-      parentHeightOffset: 0,
-      toolbar: { show: false },
-      dropShadow: {
-        top: 8,
-        blur: 3,
-        left: 3,
-        enabled: true,
-        opacity: 0.14,
-        color: theme.palette.primary.main
-      }
-    },
-    grid: {
-      show: false,
-      padding: {
-        top: -21,
-        left: -5,
-        bottom: -8
-      }
-    },
-    tooltip: { enabled: false },
-    colors: [hexToRGBA(theme.palette.primary.main, 1)],
-    markers: {
-      size: 6,
-      offsetX: -2,
-      offsetY: -1,
-      strokeWidth: 5,
-      strokeOpacity: 1,
-      colors: ['transparent'],
-      strokeColors: 'transparent'
-    },
-    stroke: {
-      width: 5,
-      curve: 'smooth',
-      lineCap: 'round'
-    },
-    xaxis: {
-      labels: { show: false },
-      axisTicks: { show: false },
-      axisBorder: { show: false }
-    },
-    yaxis: {
-      labels: { show: false }
-    }
-  }
+  const fetchQuery = useQuery({
+    ...PipelineManagerQuery.pipeline(),
+    refetchInterval: 1000
+  })
+  const [active, inactive] = partition(fetchQuery.data ?? [], p =>
+    [
+      PipelineStatus.INITIALIZING,
+      PipelineStatus.PROVISIONING,
+      PipelineStatus.RUNNING,
```

Comment:
any reason why this can't just be `PipelineStatus.RUNNING`?

Reply:
I am not sure about this - I though there are metrics incoming until pipeline is fully stopped. Should I only check for "running"?

Reply:
e.g., for PipelineStatus.INITIALIZING, PipelineStatus.PROVISIONING the stats likely won't be available yet if they get queried and probably return an error instead, just running is fine here I think -- The other states would only be active very short durations and I don't think we have clear guarantees on what happens with stats in these cases so better to just query while running.